commands/commands.go

```diff
@@ -38,137 +36,162 @@ func (h *JsonCmdHelper) Call() (interface{}, error) {
 
 var helper *JsonCmdHelper
 
-var preCfg *config.Config
-var fileCfg = config.Cfg
-
-func bindFlags() {
-	preCfg = &config.Config{}
-	RootCmd.PersistentFlags().StringVarP(&preCfg.ConfigFile, "configfile", "c", "config.toml", "config file")
-	RootCmd.PersistentFlags().StringVarP(&preCfg.DebugLevel, "debuglevel", "d", "info", "Logging level {trace, debug, info, warn, error, critical}")
-	RootCmd.PersistentFlags().StringVarP(&preCfg.AppDataDir, "appdatadir", "a", "", "wallet db path")
-	RootCmd.PersistentFlags().StringVarP(&preCfg.LogDir, "logdir", "l", "", "log data path")
-	RootCmd.PersistentFlags().StringVarP(&preCfg.Network, "network", "n", "testnet", "network")
-	RootCmd.PersistentFlags().BoolVar(&preCfg.Create, "create", false, "Create a new wallet")
-	RootCmd.PersistentFlags().StringVarP(&preCfg.QServer, "qserver", "s", "127.0.0.1:8030", "qitmeer node server")
-	RootCmd.PersistentFlags().StringVarP(&preCfg.QUser, "quser", "u", "admin", "qitmeer node user")
-	RootCmd.PersistentFlags().StringVarP(&preCfg.QPass, "qpass", "p", "123456", "qitmeer node password")
-	RootCmd.PersistentFlags().StringVarP(&preCfg.WalletPass, "pubwalletpass", "P", "public", "data encryption password")
-	RootCmd.PersistentFlags().BoolVar(&preCfg.QNoTLS, "qnotls", fileCfg.QNoTLS, "disable TLS")
-	RootCmd.PersistentFlags().StringVar(&preCfg.QCert, "qcert", fileCfg.QCert, "Certificate path")
-	RootCmd.PersistentFlags().BoolVar(&preCfg.QTLSSkipVerify, "qtlsskipverify", fileCfg.QTLSSkipVerify, "skip TLS verification")
-
-	RootCmd.PersistentFlags().Int64Var(&preCfg.Confirmations, "confirmations", 10, "Number of block confirmations ")
-	RootCmd.PersistentFlags().BoolVar(&preCfg.UI, "ui", true, "Start Wallet with RPC and webUI interface")
-	RootCmd.PersistentFlags().StringArrayVar(&preCfg.Listeners, "listeners", fileCfg.Listeners, "rpc listens")
-	RootCmd.PersistentFlags().StringVar(&preCfg.RPCUser, "rpcUser", fileCfg.RPCUser, "rpc user,default by random")
-	RootCmd.PersistentFlags().StringVar(&preCfg.RPCPass, "rpcPass", fileCfg.RPCPass, "rpc pass,default by random")
-	RootCmd.PersistentFlags().Int64Var(&preCfg.MinTxFee, "mintxfee", fileCfg.MinTxFee, "The minimum transaction fee in QIT/kB default 20000 (aka. 0.0002 MEER/KB)")
-}
+var userConf = config.Cfg
+var defaultConf = config.NewDefaultConfig()
+
+func bindFlags() error {
+	cobra.OnInitialize(initConfig)
 
-// LoadConfig config file and flags
-func LoadConfig(cmd *cobra.Command, args []string) {
-	// load configfile ane merge command ,but don't udpate configfile
-	_, err := toml.DecodeFile(preCfg.ConfigFile, fileCfg)
-	if err != nil {
-		if cmd.Flag("configfile").Changed {
-			if fExit, _ := utils.FileExists(preCfg.ConfigFile); fExit {
-				log.Error(fmt.Sprintf("config file err: %s", err))
-				return
-			}
-			return
+	dcf := config.DefaultConfigFile
+	if exists, _ := utils.FileExists(dcf); !exists {
+		if _, err := utils.FileCopy(config.SampleConfigFile, dcf); err != nil {
```

Comment:
At the beginning, if not `config.toml` exist, I think might be better to show some information to the user terminal & asking for confirmation before creating the `config.toml` from `sample-config.toml` file.

Reply:
good point, but I need to clarify a detail first, after this PR, no matter if user specifies a custom config file or not, ./config.toml will still be created since there should be an entry config file to read default flags for cobra, so if I user denied to create ./config.toml, the application would exit.

Reply:
Yes, of cause, the app will exit since there is no `config.toml` file to use if user refused to copy the `sample-config.toml`.